src/dllist.c

```diff
@@ -1077,12 +1077,18 @@ static PyObject* dllist_pop(DLListObject* self, PyObject *arg)
         self->last = (PyObject*)prev_node;
     }
 
+    if (self->last_accessed_node == (PyObject*)del_node)
```

Comment:
There is a problem with this if statement. `last_accessed_node` and `last_accessed_idx` should always point at the same node. When you remove an item from a list, all nodes behind it are shifted to a lower index, so these two fields can become unsynchronized.
The simplest solution for this issue would be to unconditionally invalidate the cache like `dllist_remove`. Alternatively you could decrement `last_accessed_idx` if it's larger than the removed index (`dllist_popleft` does something similar).